src/calliope/config/model_data_checks.yaml

```diff
@@ -29,6 +29,15 @@ fail:
   - where: storage_initial>1
     message: "storage_initial is a fraction; values larger than 1 are not allowed."
 
+  - where: storage_initial<0
+    message: "storage_initial is a fraction; values smaller than 0 are not allowed."
+
+  - where: flow_cap_min<0
+    message: "flow_cap_min has to be non-negative."
+
+  - where: (flow_cap_min AND flow_cap_max) AND flow_cap_max<flow_cap_min
```

Comment:
should this be extended to every min/max param pair?

Reply:
See the remark above. Maybe a general copy-pasted:

```yaml
where: (param_min AND param_min<LB) OR (param_max AND param_max>UB) OR (param_min AND param_max AND param_min>param_max)
message: Minimum or maximum value of parameter `param` are misconfigured.
```

That could be quickly replicated (only necessary replacements: `param`, `LB`, and `UB`) for each parameter that this would apply to, and serve as a dumb, but maybe sufficient, default check that could later on be made more explicit?

---

Alternatively, if the parser would support this, a lower- and upper-bound could be set for each parameter, which could then be automatically added to the documentation, and serve as check:

```yaml
flow_cap_min:
  default: 0
  min: 0
  max: min(.inf, flow_cap_max)
  x-type: float
  # ... other stuff
```

That would require the `min(.inf, flow_cap_max)` to gracefully handle the case that `flow_cap_max` does not exist (I suppose?).

Reply:
Sadly, the schema wouldn't allow it

Reply:
Too bad :/